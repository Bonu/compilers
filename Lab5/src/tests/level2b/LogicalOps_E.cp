
// Main: Test logical operators.

class Main{

    int i = 5;
    float a = 3.2E+5;
    bool p = true;

  /* Simple Logical Expressions */
    bool q = a && p;
    bool r = p && i;

    bool s = !i;
    bool t = !a;

    bool u = p || a;
    bool v = i || a;

  void main(){
  }
} /* end of Main */
