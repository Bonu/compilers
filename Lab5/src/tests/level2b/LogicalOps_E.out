line 11: invalid argument 1 to operator `&&'
          -- expects a boolean argument but found float
line 12: invalid argument 2 to operator `&&'
          -- expects a boolean argument but found int
line 14: unary operator `!' expects a boolean argument but found int
line 15: unary operator `!' expects a boolean argument but found float
line 17: invalid argument 2 to operator `||'
          -- expects a boolean argument but found float
line 18: invalid argument 1 to operator `||'
          -- expects a boolean argument but found int
line 18: invalid argument 2 to operator `||'
          -- expects a boolean argument but found float
7 errors were detected
